staging/src/k8s.io/apiserver/pkg/server/deleted_kinds.go

```diff
@@ -41,7 +41,16 @@ type resourceExpirationEvaluator struct {
 	serveRemovedAPIsOneMoreRelease bool
 }
 
-func newResourceExpirationEvaluator(currentVersion apimachineryversion.Info) (*resourceExpirationEvaluator, error) {
+// ResourceExpirationEvaluator indicates whether or not a resource should be served.
+type ResourceExpirationEvaluator interface {
+	// RemoveDeletedKinds inspects the storage map and modifies it in place by removing storage for kinds that have been deleted.
+	// versionedResourcesStorageMap mirrors the field on APIGroupInfo, it's a map from version to resource to the storage.
+	RemoveDeletedKinds(groupName string, versionedResourcesStorageMap map[string]map[string]rest.Storage)
```

Comment:
see comment below about `RemoveDeletedKinds`.

Reply:
> see comment below about `RemoveDeletedKinds`.

Well, you're going to be exposing something, somewhere.  This is a reasonable interface based on the rest of the package.